cpp/lib/src/FileUtil.cc

```diff
@@ -514,6 +515,29 @@ bool MakeEmpty(const std::string &path) {
 }
 
 
+void TouchFileOrDie(const std::string &path) {
+    if (Exists(path))
+        OpenForAppendingOrDie(path);
+    else
+        WriteStringOrDie(path, "");
+}
+
+
+void ChangeOwnerOrDie(const std::string &path, const std::string &user, const std::string &group, const bool recursive) {
+    std::vector<std::string> args;
+    if (recursive)
+        args.emplace_back("-R");
+
+    std::string user_and_group(user);
+    if (not group.empty())
+        user_and_group += ":" + group;
+    args.emplace_back(user_and_group);
+
+    args.emplace_back(path);
+    ExecUtil::ExecOrDie(ExecUtil::LocateOrDie("chown"), args);
```

Comment:
Do you need to use "chgrp" if there is an empty user?

Reply:
No, if the parameter given to chown starts with `:` it will only change the group.